app/views/articles/_single_story.html.erb

```diff
@@ -147,17 +147,23 @@
           <small class="crayons-story__tertiary fs-xs mr-2">
             <%= t("views.articles.reading_time", count: story.reading_time < 1 ? 1 : story.reading_time) %>
           </small>
-          <button
-            type="button"
-            id="article-save-button-<%= story.id %>"
-            class="crayons-btn crayons-btn--secondary crayons-btn--s w-max bookmark-button"
-            data-reactable-id="<%= story.id %>"
-            data-article-author-id="<%= story.user_id %>"
-            aria-label="<%= t("views.articles.save.aria_label") %>"
-            title="<%= t("views.articles.save.title") %>">
-            <span class="bm-initial"><%= t("views.articles.save.initial") %></span>
-            <span class="bm-success"><%= t("views.articles.save.success") %></span>
-          </button>
+          <% if !@user or story.user_id != @user.id %>
```

Comment:
Unfortunately I think this means on a user's profile page, the buttons will always be hidden.

If I'm logged in as "user1", and visit the profile page of "user2" - the `@user` variable here is `user2` (not me), and all their articles will be missing the save button due to this condition.

I definitely agree it would be nice to avoid the flash of the button disappearing on my own posts, but since we already have this issue in production, perhaps we should just remove this guard clause for now, and I can create a separate issue to investigate ways to avoid the flash?

Reply:
oh right I agree with that